straxen/plugins/event_processing.py

```diff
@@ -305,7 +320,7 @@ def fill_events(self, result_buffer, events, split_peaks):
     def fill_result_i(self, event, peaks):
         """For a single event with the result_buffer"""
         # Consider S2s first, then S1s (to enable allow_posts2_s1s = False)
-        largest_s2s, s2_idx = self.get_largest_sx_peaks(peaks, s_i=2)
+        largest_s2s, s2_idx = self.get_largest_sx_peaks(peaks, s_i=2, number_of_peaks=0)
```

Comment:
This is super misleading. You should put at least a comment that putting `number_of_peaks=0` means you return now all peaks not zero peaks.

Reply:
Thanks! I'll modify it accordingly.